core/src/main/scala/cats/instances/hash.scala

```diff
@@ -34,4 +36,34 @@ trait HashInstances extends kernel.instances.HashInstances {
 
     }
 
+  implicit val catsDeferForHash: Defer[Hash] = HashInstances.catsDeferForHashCache
```

Comment:
why is this a `val` rather than `def`? Since it refers to a `val` on the object, it doesn't need to be does it?

I don't know if that will help the mima issue or not. The ReverseMissingMethod problem seems like it has to do with reverse binary compatibility, which I don't think we require (using a later version works, but replacing a later version with an earlier version may break).

Reply:
Yes, switching these to `def` will fix the problems.

> the ReverseMissingMethod problem seems like it has to do with reverse binary compatibility

It may seem like it, but it doesn't. We only run MiMa in backwards-compatibility mode (not forwards-compatibility mode), so anything it flags is in fact related to backwards-compatibility.

The error message explains.
```
 [error]  * abstract method catsDeferForHash()cats.Defer in interface cats.instances.HashInstances is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]
```

`val`s in `trait`s are encoded as abstract methods that must be implemented by the `class` extending the trait. They do not have a default implementation, therefore adding them is not backwards-compatible.

Reply:
Yeah, that did it (locally at least). I can't believe I moved the instances to the object and missed the switch to `def` 🤦🏻